app/controllers/users.js

```diff
@@ -0,0 +1,29 @@
+const errors = require('../errors');
+const logger = require('../logger');
+const { encryptPassword } = require('../helpers/wcrypt');
+const { USER_CREATED } = require('../../config/constants');
+const UserServices = require('../services/users');
+
+const createUser = async (req, res, next) => {
+  try {
+    const { body: userData } = req;
+    const passwordEncrypted = await encryptPassword(userData.password);
+    if (passwordEncrypted) userData.password = passwordEncrypted;
+    await UserServices.createUser(userData);
+    return res.status(201).send({
+      message: USER_CREATED,
+      data: { name: userData.name }
+    });
+  } catch (err) {
+    logger.info('Error creating user: ', err);
+    if (err.errors) {
+      const messages = err.errors.map(e => e.message);
+      return next(errors.unprocessableEntity(messages));
+    }
+    return next(errors.unprocessableEntity(err.message));
+  }
```

Comment:
It would be better than for each operation where an exception can be raised, it has its own custom error, and here simply log the error that the catch takes and also pass it to the next function.